src/main/java/com/hillel/items_exchange/service/impl/AdvertisementServiceImpl.java

```diff
@@ -199,4 +217,47 @@ private AdvertisementTitleDto buildAdvertisementTitle(Advertisement advertisemen
                 .ownerAvatar(advertisement.getUser().getAvatarImage())
                 .build();
     }
+
+    private AdvertisementDisplayDto buildAdvertisementDisplayDto(Advertisement advertisement) {
+        Location advLocation = advertisement.getLocation();
+        long id = advertisement.getId();
+        Subcategory subcategory = advertisement.getSubcategory();
+        String createdDate = advertisement.getCreated().format(DateTimeFormatter.ofPattern(dateFormat));
+        String age = Optional.ofNullable(advertisement.getAge()).map(AgeRange::getValue).orElse("");
+        return AdvertisementDisplayDto.builder()
```

Comment:
As for me, you could use BeanUtils to copy main part of parameters and use builder for specific ones. What do you think about it?

Reply:
Great idea!